### PR TITLE
feat: fetch networks from be

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes BX-####
Figma link (if any):

with this pr we'll start fetching chains information from backend at build time

- we fetch networks into a json
- we use that file  to build the wagmi client with custom information chains
- we use that file to get additional information as which chain support swap, transactions, etc. `servicesEnabled` from the `BackendNetwork` interface
- we use that info to get native asset address
- i removed a bunch of hardcoded vars related to chains that are now dynamic

### CODE REVIEW

File: scripts/networks.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  try {
+    console.log('Fetching networks ...');
+    await fetchData();
+    console.log('Networks data fetched and available.');
+    process.exit(0);
+  } catch (error) {
+    console.error('Error fetching networks data:', error);
+    process.exit(1);
+  }
+}
```
Comment (magiziz) [inline scripts/networks.js:105]: 🤩

File: src/core/references/chains.ts [hunk 2]
```diff
@@ -1,198 +1,104 @@
...
+
+export const supportedApprovalsChainIds = filterChainIdsByService(
+  (services) => services.addys.approvals,
+);
+
+export const supportedTransactionsChainIds = filterChainIdsByService(
+  (services) => services.addys.transactions,
+);
+
+export const supportedAssetsChainIds = filterChainIdsByService(
```
Comment (magiziz) [inline src/core/references/chains.ts:90]: This is so clean!

File: src/core/resources/chains/chainMetadata.ts [hunk 3]
```diff
@@ -1,14 +1,28 @@
...
 
+export const getChainMetadataRPCUrl = async ({
+  rpcUrl,
+}: {
+  rpcUrl?: string;
+}) => {
+  if (rpcUrl && isValidUrl(rpcUrl)) {
+    const provider = new JsonRpcProvider(proxyRpcEndpoint(rpcUrl, 0));
+    const network = await provider.getNetwork();
+    return { chainId: network.chainId };
```
Comment (magiziz) [inline src/core/resources/chains/chainMetadata.ts:21]: Is it possible to use viem ?

File: src/entries/popup/hooks/useWalletName.ts [hunk 4]
```diff
@@ -2,10 +2,15 @@ import { Address } from 'viem';
...
+import { ChainId } from '~/core/types/chains';
 import { truncateAddress } from '~/core/utils/address';
 
 export const useWalletName = ({ address }: { address?: Address }) => {
-  const { data: ensName } = useEnsName({ address });
+  const { data: ensName, error } = useEnsName({
+    address,
+    chainId: ChainId.mainnet,
+  });
+  console.log('-- useWalletName', ensName, error);
```
Comment (magiziz) [inline src/entries/popup/hooks/useWalletName.ts:13]: Remove ?

**Review by magiziz — APPROVED** [review]
This is awesome! Nothing major just had some few questions 🙌👍

File: src/core/resources/chains/chainMetadata.ts (hunk 3 above)
Comment (estebanmino) [reply src/core/resources/chains/chainMetadata.ts:21]: in which sense?

this method is used when the user puts an url in the custom network form so we don't have that rpc as a network in wagmi, so this is how we're just fetching the chainid

File: src/entries/popup/hooks/useWalletName.ts (hunk 4 above)
Comment (estebanmino) [reply src/entries/popup/hooks/useWalletName.ts:13]: uh

**Pull request merged**